### PR TITLE
Spsh 935

### PR DESCRIPTION
**Pull request by jaggarnaut**
# Description
A clean installation of the client repo reveals some vulnerabilities inside npm dependencies. By fixing them (which we need to), npm run build throws a lot of typescript errors. We want to fix them inside this separate scope.

## Links to Tickets or other PRs
https://ticketsystem.dbildungscloud.de/browse/SPSH-935

# To Do
- [x] run `npm run build` and fix errors
- [x] run `npm run lint` and fix errors
- [x] manually test SPSH site for errors
- [x] fix coverage (if necessary)
- [x] fix sonar cloud errors
- [x] fix playwright (if necessary)
- [x] fix canCommit errors
- [x] Rolle cannot be edited, empty form is shown
- [x] Rolle cannot be created
- [x] ~~Person cannot be created (canCommit not true)~~ -> BE Bug, wird separat behandelt

### CODE REVIEW

File: src/components/form/RolleForm.vue [hunk 1]
```diff
@@ -22,7 +22,7 @@
     isEditActive?: boolean;
     onHandleConfirmUnsavedChanges: () => void;
     onHandleDiscard: () => void;
-    onShowDialogChange: (value: boolean) => void;
+    onShowDialogChange: (value: boolean | undefined) => void;
```
Comment (pkleybolte) [inline src/components/form/RolleForm.vue:25]:
```suggestion
    onShowDialogChange: (value?: boolean) => void;
```

File: src/views/admin/rollen/RolleCreationView.vue [hunk 2]
```diff
@@ -50,71 +58,30 @@
...
+    selectedRollenNameProps,
+    selectedMerkmale,
+    selectedMerkmaleProps,
+    selectedServiceProviders,
+    selectedServiceProvidersProps,
+    selectedSystemRechte,
+    selectedSystemRechteProps,
+  }: RolleFieldDefinitions = getRolleFieldDefinitions(formContext);
+
+  const isFormDirty: boolean = getDirtyState();
```
Comment (pkleybolte) [inline src/views/admin/rollen/RolleCreationView.vue:78]: Wird das nur ein Mal aufgerufen und nie aktualisiert?

File: src/views/admin/rollen/RolleDetailsView.vue [hunk 3]
```diff
@@ -92,67 +99,31 @@
...
+    selectedRollenNameProps,
+    selectedMerkmale,
+    selectedMerkmaleProps,
+    selectedServiceProviders,
+    selectedServiceProvidersProps,
+    selectedSystemRechte,
+    selectedSystemRechteProps,
+  }: RolleFieldDefinitions = getRolleFieldDefinitions(formContext);
+
+  const isFormDirty: boolean = !isEditActive.value ? getDirtyState() : false;
```
Comment (pkleybolte) [inline src/views/admin/rollen/RolleDetailsView.vue:125]: Wird das je aktualisiert?

File: src/views/admin/rollen/RolleDetailsView.vue [hunk 4]
```diff
@@ -393,7 +352,7 @@
             <RolleForm
               :onHandleConfirmUnsavedChanges="handleConfirmUnsavedChanges"
               :onHandleDiscard="navigateToRolleTable"
-              :onShowDialogChange="(value: boolean) => (showUnsavedChangesDialog = value)"
+              :onShowDialogChange="(value: boolean | undefined) => (showUnsavedChangesDialog = value || false)"
```
Comment (pkleybolte) [inline src/views/admin/rollen/RolleDetailsView.vue:355]:
```suggestion
              :onShowDialogChange="(value?: boolean) => (showUnsavedChangesDialog = !!value)"
```

File: src/views/admin/KlasseCreationView.vue [hunk 5]
```diff
@@ -225,7 +225,7 @@
           :showUnsavedChangesDialog="showUnsavedChangesDialog"
           :onHandleConfirmUnsavedChanges="handleConfirmUnsavedChanges"
           :onHandleDiscard="navigateToKlasseManagement"
-          :onShowDialogChange="(value: boolean) => (showUnsavedChangesDialog = value)"
+          :onShowDialogChange="(value: boolean | undefined) => (showUnsavedChangesDialog = value || false)"
```
Comment (pkleybolte) [inline src/views/admin/KlasseCreationView.vue:228]:
```suggestion
          :onShowDialogChange="(value?: boolean) => (showUnsavedChangesDialog = !!value)"
```

File: src/views/admin/KlasseDetailsView.vue [hunk 6]
```diff
@@ -228,7 +228,7 @@
               :showUnsavedChangesDialog="showUnsavedChangesDialog"
               :onHandleConfirmUnsavedChanges="handleConfirmUnsavedChanges"
               :onHandleDiscard="navigateToKlasseManagement"
-              :onShowDialogChange="(value: boolean) => (showUnsavedChangesDialog = value)"
+              :onShowDialogChange="(value: boolean | undefined) => (showUnsavedChangesDialog = value || false)"
```
Comment (pkleybolte) [inline src/views/admin/KlasseDetailsView.vue:231]:
```suggestion
              :onShowDialogChange="(value?: boolean) => (showUnsavedChangesDialog = !!value)"
```

File: src/views/admin/PersonCreationView.vue [hunk 7]
```diff
@@ -291,7 +291,7 @@
         :discardButtonLabel="$t('admin.person.discard')"
         id="person-creation-form"
         :onDiscard="navigateToPersonTable"
-        @onShowDialogChange="(value: boolean) => (showUnsavedChangesDialog = value)"
+        @onShowDialogChange="(value: boolean | undefined) => (showUnsavedChangesDialog = value || false)"
```
Comment (pkleybolte) [inline src/views/admin/PersonCreationView.vue:294]:
```suggestion
        @onShowDialogChange="(value?: boolean) => (showUnsavedChangesDialog = !!value)"
```

File: src/views/admin/PersonCreationView.vue [hunk 8]
```diff
@@ -307,7 +307,7 @@
           v-model:selectedOrganisation="selectedOrganisation"
           v-model:selectedRolle="selectedRolle"
           v-model:selectedKlasse="selectedKlasse"
-          @update:selectedOrganisation="(value: string) => (selectedOrganisation = value)"
+          @update:selectedOrganisation="(value: string | undefined) => (selectedOrganisation = value)"
```
Comment (pkleybolte) [inline src/views/admin/PersonCreationView.vue:310]:
```suggestion
          @update:selectedOrganisation="(value?: string) => (selectedOrganisation = value)"
```

File: src/views/admin/PersonDetailsView.vue [hunk 9]
```diff
@@ -1031,9 +1031,9 @@
                   v-model:selectedOrganisation="selectedOrganisation"
                   v-model:selectedRolle="selectedRolle"
                   v-model:selectedKlasse="selectedKlasse"
-                  @update:selectedOrganisation="(value: string) => (selectedOrganisation = value)"
+                  @update:selectedOrganisation="(value: string | undefined) => (selectedOrganisation = value)"
```
Comment (pkleybolte) [inline src/views/admin/PersonDetailsView.vue:1034]:
```suggestion
                  @update:selectedOrganisation="(value?: string) => (selectedOrganisation = value)"
```

File: src/views/admin/PersonDetailsView.vue [hunk 10]
```diff
@@ -1031,9 +1031,9 @@
...
                   @update:selectedRolle="(value: string | undefined) => (selectedRolle = value)"
-                  @update:selectedKlasse="(value: string) => (selectedKlasse = value)"
+                  @update:selectedKlasse="(value: string | undefined) => (selectedKlasse = value)"
```
Comment (pkleybolte) [inline src/views/admin/PersonDetailsView.vue:1036]:
```suggestion
                  @update:selectedKlasse="(value?: string) => (selectedKlasse = value)"
```

File: src/views/admin/rollen/RolleDetailsView.vue (hunk 3 above)
Comment (YoussefBouch) [reply src/views/admin/rollen/RolleDetailsView.vue]: Habe es jetzt gefixed. Dirtiness is reaktiv.

File: src/views/admin/rollen/RolleCreationView.vue [hunk 11]
```diff
@@ -328,7 +283,7 @@
           :administrationsebenen="administrationsebenen"
           :onHandleConfirmUnsavedChanges="handleConfirmUnsavedChanges"
           :onHandleDiscard="navigateToRolleManagement"
-          :onShowDialogChange="(value: boolean) => (showUnsavedChangesDialog = value)"
+          :onShowDialogChange="(value: boolean | undefined) => (showUnsavedChangesDialog = value || false)"
```
Comment (pkleybolte) [inline src/views/admin/rollen/RolleCreationView.vue:286]:
```suggestion
          :onShowDialogChange="(value?: boolean) => (showUnsavedChangesDialog = !!value)"
```

**Pull request merged**